packages/gatsby/src/bootstrap/load-plugins/__tests__/load-plugins.js

```diff
@@ -71,4 +71,76 @@ describe(`Load plugins`, () => {
 
     expect(plugins).toMatchSnapshot()
   })
+
+  describe(`TypeScript support`, () => {
+    it(`loads gatsby-plugin-typescript if not provided`, async () => {
+      const config = {
+        plugins: [],
+      }
+
+      let plugins = await loadPlugins(config)
+
+      plugins = replaceFieldsThatCanVary(plugins)
+
+      expect(plugins).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            browserAPIs: [],
+            id: ``,
+            name: `gatsby-plugin-typescript`,
+            nodeAPIs: [
+              `resolvableExtensions`,
+              `onCreateBabelConfig`,
+              `onCreateWebpackConfig`,
+            ],
+            pluginOptions: {
+              plugins: [],
+            },
+            resolve: ``,
+            ssrAPIs: [],
+            version: `1.0.0`,
+          }),
+        ])
+      )
+    })
+
+    it(`uses the user provided plugin-typescript if provided`, async () => {
+      const config = {
+        plugins: [
+          {
+            resolve: `gatsby-plugin-typescript`,
+            options: {
+              jsxPragma: `h`,
+            },
+          },
+        ],
+      }
+
+      let plugins = await loadPlugins(config)
+
+      plugins = replaceFieldsThatCanVary(plugins)
+
+      expect(plugins).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            browserAPIs: [],
+            id: ``,
+            name: `gatsby-plugin-typescript`,
+            nodeAPIs: [
+              `resolvableExtensions`,
+              `onCreateBabelConfig`,
+              `onCreateWebpackConfig`,
+            ],
+            pluginOptions: {
+              plugins: [],
+              jsxPragma: `h`,
+            },
+            resolve: ``,
+            ssrAPIs: [],
+            version: `1.0.0`,
+          }),
+        ])
+      )
```

Comment:
I would like to see additional assertion that make sure we have only 1 instance of typescript plugin - touches on comment that @wardpeet raised in the `load.js` source code

Reply:
Good call. I added it and it raised a bug that I fixed